Base.rte/Scenes/Objects/Bunkers/BunkerSystems/Docks/DropshipDock.lua

```diff
@@ -1,24 +1,43 @@
+function Create(self)
+	self.healTimer = Timer();
+end
+
 function Update(self)
-    for actor in MovableMan.Actors do
-	--See if a dropship is within 45 pixel range of the docking unit and if it has the AI mode set to "Stay".
-	if (actor.ClassName == "ACDropShip") and (actor.AIMode == Actor.AIMODE_STAY) and (math.abs(actor.Pos.X - self.Pos.X) < 45) and (math.abs(actor.Pos.Y - self.Pos.Y) < 45) then
-	    --[[ WARNING, AUTOSCUTTLE WILL KICK IN ONCE THE DS NOTICES IT'S NOT MOVING ANYWHERE.
-	    --Pin the ship and place it nicely in the docking unit.
-	    actor.Vel = Vector(0,0);
-	    actor.Pos = Vector(self.Pos.X + 0, self.Pos.Y + 38);  --replace offsets with values that place the dropship in a nice position
-	    --]]
-	    --TEMPORARY SOLUTION. GRAVITATE AT DOCK PIXELS.
-	    local diff = math.sqrt(math.pow((actor.Pos.X-self.Pos.X),2) + math.pow((actor.Pos.Y - (self.Pos.Y + 38)),2))
-	    if diff < 5 then
-		diff = 5
-	    end
-	    if diff < 85 then
-		local diffx = actor.Pos.X - self.Pos.X;
-		local diffy = actor.Pos.Y - (self.Pos.Y + 38);
-		local ang = math.atan2(diffy,diffx);
-		actor.Vel.Y = actor.Vel.Y - (6 / diff * math.sin(ang));
-		actor.Vel.X = actor.Vel.X - (3 / diff * math.cos(ang));
-	    end
+
+	--Settle into terrain instead of gibbing
+	if self.WoundCount > (self.GibWoundLimit * 0.9) then
+		self.ToSettle = true;
+	end
+
+	if self.craft and IsActor(self.craft) and self.craft.AIMode == Actor.AIMODE_STAY then
+		--Disable collisions with the ship
+		self.craft:SetWhichMOToNotHit(self, 100);
+		
+		--Pin the ship and pull it nicely into the docking unit.
+		local dist = SceneMan:ShortestDistance(self.craft.Pos, self.Pos + Vector(0, self.craft.Radius / 2), SceneMan.SceneWrapsX);
+		self.craft.Vel = self.craft.Vel * 0.9 + dist / (3 + self.craft.Vel.Magnitude);
+		self.craft.AngularVel = self.craft.AngularVel * 0.9 - self.craft.RotAngle * 3;
+		
+		if self.craft.Status < 3 then
+			self.craft.Status = 1;	--Deactivated
+		end
+
+		--Heal the craft.
+		if self.healTimer:IsPastSimMS(self.craft.Mass) then
+			self.healTimer:Reset();
+			if self.craft.TotalWoundCount > 0 then
+				self.craft.Health = math.min(self.craft.Health + self.craft:RemoveAnyRandomWounds(1), self.craft.MaxHealth);
+			elseif self.craft.Health < self.craft.MaxHealth then
+				self.craft.Health = math.min(self.craft.Health + 1, self.craft.MaxHealth);
+			end
+		end
+	else
+		self.craft = nil;
+		for actor in MovableMan.Actors do
```

Comment:
This shouldn't be done every frame for expense reasons, either do it every few frames or coroutine it.
Also, even though it probably shouldn't be changed too much, it's usually got to avoid magic numbers like the 50 pixel thing here. Put it in a self variable instead.